flake.nix

```diff
@@ -1,34 +1,92 @@
 {
-    inputs = {
-        # Use unstable for flakes
-        nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
-        
-        # Add home-manager
-        home-manager = {
-            url = "github:nix-community/home-manager";
-            inputs.nixpkgs.follows = "nixpkgs";
-        };
+  inputs = {
+    # Use unstable for flakes
+    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
 
-        # Use nixos generators for generating an iso
-        nixos-generators.url = "github:nix-community/nixos-generators";
-        nixos-generators.inputs.nixpkgs.follows = "nixpkgs";
+    # Add home-manager
+    home-manager = {
+      url = "github:nix-community/home-manager";
+      inputs.nixpkgs.follows = "nixpkgs";
     };
 
-    outputs = inputs@{ self, nixpkgs, home-manager, flake-utils, ... }: {
-        ## nix build .#console
-        packages.x86_64-linux.console = inputs.nixos-generators.nixosGenerate {
-            system = "x86_64-linux";
-            format = "raw";
-            specialArgs = {
-                inherit inputs;
-            };
-            modules = [
-                ./images/common.nix
-                ./images/console.nix
-                {
-                    system.stateVersion = "23.11";
-                }
-            ];
+    # Use nixos generators for generating an iso
+    nixos-generators.url = "github:nix-community/nixos-generators";
+    nixos-generators.inputs.nixpkgs.follows = "nixpkgs";
+  };
+
+  outputs = inputs@{ self, nixpkgs, home-manager, flake-utils, ... }:
+    let
+      lib = nixpkgs.lib;
+      system = "x86_64-linux";
+
+    in
+
+    {
+      inherit lib;
+
+      # For nixos-rebuild
+      nixosConfigurations = {
+        console = lib.nixosSystem {
+          system = "x86_64-linux";
+          specialArgs = {
+            inherit self inputs system;
+          };
+          modules = [
+            ./images/console
+            ./images/common.nix
+            {
+              system.stateVersion = "23.11";
+            }
+          ];
+        };
+
+        contestant = lib.nixosSystem {
+          system = "x86_64-linux";
+          specialArgs = {
+            inherit self inputs system;
+          };
+          modules = [
+            ./images/contestant
+            ./images/common.nix
+            ./hosts/contestant/hardware-configuration.nix
```

Comment:
I was always afraid to add a hardware-configuration. Now that it defines some sort of `qemu-guest` thing, does that not interfere with running it bare metal?

Reply:
For now it is only added when you run nixos-rebuild once the system is actually build. So once we run it on bare-metal, we can remove it. (im not even sure if it is really needed right now)

Reply:
But if no hardware-configuration is included, rebuilding is not possible because you need to define your root filesystem and bootloader location